File: verba-data/src/main/java/ru/nsu/ccfit/verba/verbadata/platform/exeption/BasicVerbaException.java
package ru.nsu.ccfit.verba.verbadata.platform.exeption;

import lombok.Getter;
import ru.nsu.ccfit.verba.verbadata.platform.enumeration.Code;



@Getter
public abstract class BasicVerbaException extends RuntimeException {
    private final Code code;

    protected BasicVerbaException(Code code) {
        this(code, code.getErrorMessage());
    }

    protected BasicVerbaException(Code code, String message) {
        super(message);
        this.code = code;
    }
}

File: verba-api/src/main/kotlin/ru/nsu/ccfit/verba/verbaapi/platform/exception/BasicVerbaException.kt
package ru.nsu.ccfit.verba.verbaapi.platform.exception

import ru.nsu.ccfit.verba.verbaapi.platform.Code


abstract class BasicVerbaException protected constructor(errorCode: Code, message: String) :
    RuntimeException(message) {
    val errorCode: Code

    protected constructor(errorCode: Code) : this(errorCode, errorCode.errorMessage)

    init {
        this.errorCode = errorCode
    }
}
